sdk/src/signer/keypair.rs

```diff
@@ -44,6 +44,11 @@ impl Keypair {
 
     /// Recovers a `Keypair` from a byte array
     pub fn from_bytes(bytes: &[u8]) -> Result<Self, ed25519_dalek::SignatureError> {
+        if bytes.len() < ed25519_dalek::KEYPAIR_LENGTH {
```

Comment:
I know it doesn't matter for slice indexing, but is there any reason this should be catching longer byte slices as well? (ie `!=`)

Reply:
i considered it, but decided to punt under the impression that someone is likely abusing the poorly defined api. clearly this method signature should be
```rust
fn from_bytes(bytes: &[u8; 64]) -> Result<Self, NotTheDamnDependencysError>
```
but here we are